DOC-7960 Prometheus Changes

ibsoln opened the pull request:
https://issues.couchbase.com/browse/DOC-7960

JRascagneres commented on modules/ROOT/assets/attachments/prometheus-rules.yaml:
```diff
@@ -0,0 +1,40 @@
+        labels:
+          severity: warning
+        annotations:
+          summary: High Channel Cache Miss Rate
```
These examples have since been modified for 2.8.1 (and up) to update a couple of the existing rules and to add more examples. Should be able to just copy over from here: https://github.com/couchbase/sync_gateway/blob/master/examples/prometheus/rules/sync-gateway.rules.yml

ibsoln replied:
Done, thanks

JRascagneres commented on modules/ROOT/assets/attachments/sync-gateway-metrics.yaml:
```diff
@@ -49,11 +49,10 @@ paths:
-
-        For more details on the metrics collected and reported by Sync Gateway -- see[Sync Gateway Statistics Schema](./../stats-monitoring.html).
+        - For more details on the metrics collected and reported by Sync Gateway -- see: [Sync Gateway Statistics Schema](./../stats-monitoring.html).
+        - For more on monitoring Sync Gateway with Prometheus -- see: our blog on [Monitoring Sync Gateway with Prometheus and Grafana](https://blog.couchbase.com/monitoring-sync-gateway-prometheus-grafana/)
```
Perhaps this could be updated to the more recent one? 
https://blog.couchbase.com/monitoring-and-visualization-of-couchbase-sync-gateway-with-prometheus-and-grafana/

ibsoln replied:
Done, thanks

JRascagneres replied:
Doesn't seem done here?

JRascagneres commented on modules/ROOT/pages/_partials/prometheus-activation.adoc:
```diff
@@ -0,0 +1,18 @@
+//
+To activate the Metrics REST API you need to edit your `sync-gateway-config.json` configuration file.
+
+Include the `metricsInterface` setting and define the Sync Gateway URL and the port you want the API served on (default: 4986) -- see: <<ex-activate>>.
```
4986 is not the default really. Unless specified the metrics endpoint is not accessible like you mention in the first sentence.

ibsoln replied:
Amended, thanks.

JRascagneres commented on modules/ROOT/pages/start/gs-sgw-api-access.adoc:
```diff
@@ -12,13 +12,13 @@ include::partial$block-abstract.adoc[]
-
-include::partial$block-prometheus.adoc[]
+Sync Gateway REST APIs are accessed on different TCP ports.
+This makes it easy to expose the Public REST API (port 4984) and Metrics REST API (port 4986) to endpoints while keeping the Admin REST API (port 4985) secure behind your firewall.
```
This reads to me like the metrics endpoint is available on 4986 but you have the option to change it. Should make clear that this is not exposed unless the option is specified.

ibsoln replied:
Amended

JRascagneres commented on modules/ROOT/pages/stats-prometheus.adoc:
```diff
@@ -0,0 +1,100 @@
+--
+The provided `prometheus.yml` file specifies the configuration required to scrape the Sync{nbsp}Gateway metrics target.
+It defines Sync Gateway's `metricsInterface` as being accessible on sync_gateway:4986/_metrics.
+If you have multiple Sync Gateways, you can either statically specify all their endpoints or configure Prometheus to discover the targets.
```
Not too sure exactly what this bit means. If you want prometheus to find multiple targets you have to specify them as targets in the yml file.

ibsoln replied:
Removed discovery reference

JRascagneres reviewed (changes requested):
Couple bits to update. Main concern is in a few places it seems to suggest that the endpoint is exposed by default and then in other bits (correctly) says its not.

ibsoln committed:
DOC-7960 Prometheus Changes

https://issues.couchbase.com/browse/DOC-7960

ibsoln committed:
DOC-7960 Prometheus Changes (Post-Review Changes)

https://issues.couchbase.com/browse/DOC-7960

ibsoln committed:
DOC-7960-281 -- Prometheus Changes (config)
https://issues.couchbase.com/browse/DOC-7962

ibsoln commented:
Hi @JRascagneres if you are ok with the changes now can you approve please?

This branch is now included in the feature branch https://github.com/couchbase/docs-sync-gateway/pull/416 which is staged at https://ibsoln.github.io/stage/release281-0302/sync-gateway/2.8/sgw-whatsnew.html

JRascagneres commented on modules/ROOT/assets/attachments/sg.yaml:
```diff
@@ -16,7 +16,7 @@ properties:
 
       You can make the developer preview visible by setting the endpoint address.
```
Not sure if this is versioned but this mentions Developer Preview. Will this only show in 2.8?

JRascagneres replied:
But then again I'd say the same if we moved

JRascagneres reviewed (changes requested):
These couple questions. Otherwise looks good :)

ibsoln committed:
Remove developer preview reference

ibsoln committed:
Link to updated blog 

Monitoring and Visualization of Couchbase Sync Gateway with Prometheus and Grafana

ibsoln commented:
> These couple questions. Otherwise looks good :)

Fixed queries in latest commits

ibsoln committed:
DOC-7960 -- metricsInerface ON by default

https://issues.couchbase.com/browse/DOC-7960

Incorporate CBG-1313 changes to make metricsInterface on, on port 4986, by default.

ibsoln committed:
DOC-7960 -- Add link to metricsInterface from admin_xpvar

https://issues.couchbase.com/browse/DOC-7960

xpvar available in metrics and admin API
